edb/pgsql/compiler/relgen.py

```diff
@@ -1493,6 +1493,7 @@ def process_set_as_subquery(
 
 
 register_get_rvar(irast.Stmt)(process_set_as_subquery)
+register_get_rvar(irast.MaterializedExpr)(process_set_as_subquery)
```

Comment:
Why cannot we declare these two as decorators?

Reply:
"Because mypy doesn't support higher-kinded polymorphism"

It would narrow the type of the function, and this function gets called directly, and because of not having the higher-kinded polymorphism we can't give the decorator the type we'd need to have it work the way we want and also enforce the correct type for the function.